fili-core/src/main/java/com/yahoo/bard/webservice/util/IntervalUtils.java

```diff
@@ -209,4 +211,20 @@ public static long getTotalDuration(Collection<Interval> intervals) {
                 .map(Interval::getStart)
                 .reduce(Utils::getMinValue);
     }
+
+    /**
+     * Returns the coarsest ZonedTimeGrain among a set of ZonedTimeGrains.
+     * <p>
+     * If the set of ZonedTimeGrains is empty, return null.
```

Comment:
Why are we returning `null` here? Do we really expect `null` to be a valid return value? If we do, we should make that explicit in the return type by using `Optional` and _force_ callers to deal with the fact that it could come back as null.